### PR TITLE
Projector::unrpoject() and show click position on the map

### PR DESCRIPTION
**Pull request by noktoborus**
made in opposite to https://github.com/podusowski/walkers/pull/125

### CODE REVIEW

File: demo/src/plugins.rs [hunk 1]
```diff
@@ -78,3 +78,40 @@ impl Plugin for CustomShapes {
         );
     }
 }
+
+#[derive(Default, Clone)]
+pub struct ClickWatcher {
+    pub clicked_at: Option<Position>,
+}
+
+impl ClickWatcher {
```
Comment (podusowski) [inline demo/src/plugins.rs:87]: Needed to time to figure out how does this stuff work, and honestly, I still don't quite follow it. :)

It handles events i handle_event, then in the next frame, it clones itself, and uses the data from the event to draw the circle?

How about something like that?

```rust
#[derive(Default)]
pub struct ClickWatcher {
    pub clicked_at: Option<Position>,
}

impl ClickWatcher {
    pub fn show_my_position(&self, ui: &egui::Ui) {
        if let Some(clicked_at) = self.clicked_at {
            egui::Window::new("My Position")
                .collapsible(false)
                .resizable(false)
                .title_bar(false)
                .anchor(egui::Align2::CENTER_BOTTOM, [0., -10.])
                .show(ui.ctx(), |ui| {
                    ui.label(format!("{:.04} {:.04}", clicked_at.lon(), clicked_at.lat()))
                        .on_hover_text("my position");
                });
        }
    }
}

impl Plugin for &mut ClickWatcher {
    fn draw(&mut self, response: &Response, painter: Painter, projector: &Projector) {
        if !response.changed() && response.clicked_by(egui::PointerButton::Primary) {
            if let Some(offset) = response.hover_pos().map(|p| p - response.rect.center()) {
                self.clicked_at = Some(projector.unproject(offset));
            }
        }

        if let Some(position) = self.clicked_at {
            painter.circle_filled(projector.project(position).to_pos2(), 5.0, Color32::BLUE);
        }
    }
}
```

Whole logic is now in a single frame. `Plugin::draw` needs to be changed to take `&mut self`, but it's all right I guess.

Comment (noktoborus) [reply demo/src/plugins.rs:87]: Not agree: then the meaning of the function name is violated.

But we can add something like 

```
trait Plugin {
   ...
   fn handle(&mut self, response: &Response, projector: &Projector) {}
}
````
to use optional trait's member

Comment (podusowski) [reply demo/src/plugins.rs:87]: Yeah, fair point. Then maybe it's just `run`?

Comment (noktoborus) [reply demo/src/plugins.rs:87]: I've updated this branch, can you check?

Comment (noktoborus) [reply demo/src/plugins.rs:87]: code is checked, works fine

Comment (noktoborus) [reply demo/src/plugins.rs:87]: if ok, I want a suggestion for better naming

Comment (podusowski) [reply demo/src/plugins.rs:87]: I still think it would be better to have a single function - `run`.

File: demo/src/plugins.rs [hunk 2]
```diff
@@ -78,3 +78,46 @@ impl Plugin for CustomShapes {
...
+    pub fn handle_event(&mut self, response: &Response, projector: &Projector) {
+        if !response.changed() && response.clicked_by(egui::PointerButton::Primary) {
+            if let Some(offset) = response.hover_pos().map(|p| p - response.rect.center()) {
+                self.clicked_at = Some(projector.unproject(offset));
+            }
+        }
+    }
```
Comment (podusowski) [inline demo/src/plugins.rs:94]: This is unused, isn't it? I wonder why CI didn't catch it.

File: demo/src/plugins.rs [hunk 3]
```diff
@@ -78,3 +78,46 @@ impl Plugin for CustomShapes {
...
+
+    pub fn show_my_position(&self, ui: &egui::Ui) {
```
Comment (podusowski) [inline demo/src/plugins.rs:96]: It's rather "last clicked position".

**Review by podusowski — CHANGES_REQUESTED** [review]

**Pull request merged**